src/middlewared/middlewared/plugins/kubernetes_linux/update.py

```diff
@@ -314,12 +316,24 @@ async def route_interface_choices(self):
         )
 
     @private
-    async def validate_k8s_setup(self):
+    async def validate_k8s_setup(self, raise_exception=True):
+        error = None
         k8s_config = await self.middleware.call('kubernetes.config')
         if not k8s_config['dataset']:
-            raise CallError('Please configure kubernetes pool.')
-        if not await self.middleware.call('service.started', 'kubernetes'):
-            raise CallError('Kubernetes service is not running.')
+            error = 'Please configure kubernetes pool.'
+        if not error and not await self.middleware.call('service.started', 'kubernetes'):
+            error = 'Kubernetes service is not running.'
+
+        if not error:
+            try:
+                async with api_client({'node': True}, {'request_timeout': 2}) as (api, context):
```

Comment:
To confirm, have we tested if 2 seconds is reasonable ?

Reply:
its almost instant if service is up and serving since its local network – 2000ms should be fine :)

Reply:
Great, thanks for confirming!